src/H5Pdxpl.c

```diff
@@ -2386,3 +2496,248 @@ H5Pset_dataset_io_hyperslab_selection(hid_t plist_id, unsigned rank, H5S_seloper
+        /* Encode modify write buf property.  Use "!!" so we always get 0 or 1 */
+        *(*pp)++ = (uint8_t)(!!(*modify_write_buf));
+
+    /* Size of modify write buf property */
+    (*size)++;
+
+    FUNC_LEAVE_NOAPI(SUCCEED)
+} /* end H5P__dxfr_modify_write_buf_enc() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5P__dxfr_modify_write_buf_dec
+ *
+ * Purpose:     Callback routine which is called whenever the modify write
+ *              buffer property in the dataset transfer property list is
+ *              decoded.
+ *
+ * Return:      Success:        Non-negative
+ *              Failure:        Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5P__dxfr_modify_write_buf_dec(const void **_pp, void *_value /*out*/)
+{
+    hbool_t        *modify_write_buf = (hbool_t *)_value; /* Modify write buffer */
+    const uint8_t **pp               = (const uint8_t **)_pp;
+
+    FUNC_ENTER_PACKAGE_NOERR
+
+    /* Sanity checks */
+    HDassert(pp);
+    HDassert(*pp);
+    HDassert(modify_write_buf);
+
+    /* Decode selection I/O mode property */
+    *modify_write_buf = (hbool_t) * (*pp)++;
+
+    FUNC_LEAVE_NOAPI(SUCCEED)
+} /* end H5P__dxfr_modify_write_buf_dec() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Pset_modify_write_buf
+ *
+ * Purpose:     Allows the library to modify the contents of the write
+ *              buffer
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Pset_modify_write_buf(hid_t plist_id, hbool_t modify_write_buf)
+{
+    H5P_genplist_t *plist;               /* Property list pointer */
+    herr_t          ret_value = SUCCEED; /* Return value */
+
+    FUNC_ENTER_API(FAIL)
+    H5TRACE2("e", "ib", plist_id, modify_write_buf);
+
+    /* Check arguments */
+    if (plist_id == H5P_DEFAULT)
+        HGOTO_ERROR(H5E_PLIST, H5E_BADVALUE, FAIL, "can't set values in default property list")
+
+    if (NULL == (plist = H5P_object_verify(plist_id, H5P_DATASET_XFER)))
+        HGOTO_ERROR(H5E_PLIST, H5E_BADTYPE, FAIL, "not a dxpl")
+
+    /* Set the selection I/O mode */
+    if (H5P_set(plist, H5D_XFER_MODIFY_WRITE_BUF_NAME, &modify_write_buf) < 0)
+        HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "unable to set value")
+
+done:
+    FUNC_LEAVE_API(ret_value)
+} /* end H5Pset_modify_write_buf() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Pget_modify_write_buf
+ *
+ * Purpose:     Retrieves the "modify write buffer" property
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Pget_modify_write_buf(hid_t plist_id, hbool_t *modify_write_buf /*out*/)
```

Comment:
Should check for H5P_DEFAULT and set plist_id to correct library default DXPL if that's passed in